### ISSUE TITLE
Allow $let: $if without else

### ISSUE THREAD

**Issue by mitchhentges** [taskcluster/json-e#262]
For the following template:
```
$let:
  a: {$if: 'c == "item"', then: 'value'}
in:
  a: ${a}
```
If `c` is not "item", then the template fails to compile (and with a bit of a misleading error message):

> interpolation of 'a' produced an array or object

This can be worked around by adding a blank `else`:

```
$let:
  a: {$if: 'c == "item"', then: 'value', else: ''}
in:
  a: ${a}
```

Instead, it would be ideal if you could omit the `else`, like you can when using `else` in other situations.
When this case is triggered, the variable you specified should remain unset.

-----

Example of omitting the `else` in a situation other than `$let`:
```
a:
  $if: 'c == "item"'
  then: value
```

In this case, if `c` is not "item", then the result is just `{}`.

**Comment by Nicrii** [comment]
@owlishDeveloper I’d like to work on this issue

**Comment by owlishDeveloper** [comment]
Fixed in #323
